### ISSUE TITLE
bumpalo v3.15.3` cannot be built

### ISSUE THREAD

**Issue by zhshang1221** [jup-ag/jupiter-cpi#7]
Seems the most recent commit only support solana-program < 15? which will render many conflicts when anchor build. E.g. error: package `bumpalo v3.15.3` cannot be built because it requires rustc 1.73.0 or newer, while the currently active rustc version is 1.72.0-dev
Either upgrade to rustc 1.73.0 or newer, or use
cargo update -p bumpalo@3.15.3 --precise ver
![4901711860369_ pic](https://github.com/jup-ag/jupiter-cpi/assets/54016482/e0f99b16-db1e-41e1-a16a-7bfee1e51b2e)
![4911711860380_ pic](https://github.com/jup-ag/jupiter-cpi/assets/54016482/ced599d5-121d-45f2-a421-893520173af9)

**Comment by Kelvin-1013** [comment]
No, it is build successfully.  I used anchor version =0.29.0 and changed 
from   :   use solana_program::pubkey;
to       :   use anchor_lang::prelude::Pubkey;
previous error : 
`
error[E0308]: mismatched types
  --> src\lib.rs:46:5
   |
37 | pub fn find_jupiter_token_ledger() -> Pubkey {
   |                                       ------ expected `anchor_lang::prelude::Pubkey` because of return type
...
46 |     *token_ledger.unwrap()
   |     ^^^^^^^^^^^^^^^^^^^^^^ expected `anchor_lang::prelude::Pubkey`, found `solana_program::pubkey::Pubkey`
   |
   = note: `solana_program::pubkey::Pubkey` and `anchor_lang::prelude::Pubkey` have similar names, but are actually distinct types
note: `solana_program::pubkey::Pubkey` is defined in crate `solana_program`
  --> C:\Users\pc\.cargo\registry\src\index.crates.io-6f17d22bba15001f\solana-program-2.0.21\src\pubkey.rs:96:1
   |
96 | pub struct Pubkey(pub(crate) [u8; 32]);
   | ^^^^^^^^^^^^^^^^^
note: `anchor_lang::prelude::Pubkey` is defined in crate `solana_program`
  --> C:\Users\pc\.cargo\registry\src\index.crates.io-6f17d22bba15001f\solana-program-1.18.26\src\pubkey.rs:88:1
   |
88 | pub struct Pubkey(pub(crate) [u8; 32]);
   | ^^^^^^^^^^^^^^^^^
   = note: perhaps two different versions of crate `solana_program` are being used?

For more information about this error, try `rustc --explain E0308`.
error: could not compile `jupiter-cpi` (lib) due to 1 previous error
`
